From 10717b9c4bb75dacb1489d7b7b59d1a63bbc6025 Mon Sep 17 00:00:00 2001
From: Matthias Eliasson <matthias.eliasson@gmail.com>
Date: Tue, 22 Feb 2022 22:01:28 +0100
Subject: [PATCH] Don't generate appstream meta data on older versions

Support for NEWS file to metainfo conversion as added in version 0.12.9
of appstreamcli. So don't generate metadata info for older versions and
provide a file instead.
---
 data/meson.build | 29 -----------------------------
 1 file changed, 29 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index 9b2ea4b2..958e05ce 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -30,35 +30,6 @@ if desktop_file_validate.found()
   )
 endif
 
-# Merge release information into MetaInfo file
-tilix_metainfo_name = '@0@.appdata.xml'.format(project_id)
-appstreamcli = find_program('appstreamcli')
-metainfo_with_releases = custom_target('metainfo-news-merge',
-    input : ['../NEWS', 'metainfo/@0@.in'.format(tilix_metainfo_name)],
-    output : ['untranslated-@0@.appdata.xml'.format(project_id)],
-    command : [appstreamcli, 'news-to-metainfo', '--limit=6', '@INPUT0@', '@INPUT1@', '@OUTPUT@']
-)
-
-# Install the MetaInfo file
-metainfo_file = i18n.merge_file(
-    output: tilix_metainfo_name,
-    input: metainfo_with_releases,
-    po_dir: meson.project_source_root() / 'po',
-    install: true,
-    install_dir: datadir / 'metainfo'
-)
-
-# Validate MetaInfo file
-if appstreamcli.found()
-  test (
-    'Validate metainfo file',
-    appstreamcli,
-    args: ['validate', '--no-net',
-           join_paths(meson.current_source_dir(), 'metainfo', '@0@.in'.format(tilix_metainfo_name))
-    ]
-  )
-endif
-
 # Build resources
 gresource = gnome.compile_resources(
     'tilix',
-- 
2.35.1

